lib/chef/resource/msu_package.rb

```diff
@@ -24,18 +24,14 @@ class Resource
     class MsuPackage < Chef::Resource::Package
       include Chef::Mixin::Uris
 
+      resource_name :msu_package
       provides :msu_package, os: "windows"
 
       allowed_actions :install, :remove
-
-      def initialize(name, run_context = nil)
-        super
-        @resource_name = :msu_package
-        @source = name
-        @action = :install
-      end
+      default_action :install
 
       property :source, String,
+                name_property: true,
```

Comment:
This fixes 1 of 2 bugs in how we setup this resource. I've filed #6674 for the other issue, which I'll fix and get a test around